.github/workflows/run-tests.yml

```diff
@@ -8,7 +8,7 @@ concurrency:
     group: ${{ github.workflow}}-${{ github.ref }}
     cancel-in-progress: true
 jobs:
-  unit-test:
+  check-dependencies:
```

Comment:
I thought you bound the dependency:analyze-only to a phase after test-compile, so it runs during `mvn test`. In that case we would not need separate jobs. I do like having explicit jobs for this but we worked hard to get the build stable at around 15min. Adding ~5min back on the build doesnt feel good 😬 So if we can run the check before our tests but within the test jobs, that would hopefully reduce that time to its necessary minimum. As we wouldnt setup JDK, cache, compile again https://github.com/dhis2/dhis2-core/pull/10304#pullrequestreview-925471526

Reply:
I agree with a separate job check-dependencies clearly highlighting at what stage the build failed.

We are adding 4.5min though:
![Screenshot from 2022-03-30 15-31-51](https://user-images.githubusercontent.com/4661144/160846642-ff181bcb-c8dc-44e9-8fab-3c4263eacbb5.png)

I value time more than the above. I don't think the benefit of the dedicated job warrants the added time. If a test fails for example, I need to check which test failed. Same with, dependency checks failing the build, I need to check which dependencies caused the build to fail. Thus, navigate to the build logs to see the error. If we are on a tie on this one 😋 I would let the backend team meeting decide.

Reply:
Oh, so your argument is we run formatting before test jobs in either case (with separate dependency-check or not). So you calculate check-dependencies duration - check-formatting duration. We still have 3.5min added time. Since you said the actual time it takes to check dependencies is fast, most of it must be spent setting up JDK, cache and compiling. Since we already do that repeatedly throughout our workflows I would rather not do that again with check-dependencies. Thus, include that in the test jobs

Reply:
In this whole process most of the time is used to compile source and compile the tests.
We do it again in every job.
Anyway, I agree that we can remove the check-dependencies job as it is included in the other test jobs.
I would think to put check-formatting back to a separate workflow then. Formatting errors are not so frequent. Do we want one more minute in the build over sometimes not running the test jobs?

Reply:
hard to say 😬 

having test jobs depend on the check-formatting 
pro: we save 3 jobs from being taken of the 20 concurrent job per organization quota in case formatting is wrong
con: we add 1min to our build time for all the builds where formatting is ok

If we assume like you said formatting is mostly ok, then keeping it as a separate workflow/job running in parallel to the tests is probably best. So I would also go back to that. Maybe you can add your improvements into it though. You did adapt it a bit to get rid of the longer JDK post setup step 😄